specs/dotsigstore_bundle/pb/bundle.proto

```diff
@@ -0,0 +1,167 @@
+        // This is primarily used to verify the
+        // integration timestamp's value.
+        string inclusion_promise = 5;
+        // The inclusion proof can be used for online verification that the
+        // entry was appended to the log, and that the log has not been
+        // altered.
+        InclusionProof inclusion_proof = 6;
+}
+
+// A list of base64 encoded RFC3161 signed timestamps.
+message SignedTimestamps {
+        repeated string signed_timestamps = 1;
+}
+
+// This message contains the transparency log entry (inclusion promise and
+// proof) and signed timestamps, provided by sigstore and/or the user.
+message TimestampedTransparencyLogEntry {
+        TransparencyLogEntry tlog_entry = 1;
+        SignedTimestamps timestamps = 2;
+}
+
+message VerificationData {
+        oneof data {
+                // This is the regular inclusion promise and proof, where
+                // the timestamp is coming from the transparency log
+                TransparencyLogEntry tlog_entry = 1;
+                // This is the regular inclusion promise and proof, where
+                // the timestamp is coming from both the transparency log,
+                // and RFC3161 timestamps provided by Sigstore and user.
+                TimestampedTransparencyLogEntry timestamped_tlog_entry = 2;
+                // A list of RFC3161 signed timestamps provided by the user.
+                // This can be used when the entry has not been stored on a
+                // transparency log.
+                SignedTimestamps timestamps = 4;
+        }
+}
+
+// PublicKeyIdentifier can be used to identify an (out of band) delivered
+// key, to verify a signature.
+message PublicKeyIdentifier {
+        // Optional unauthenticated hint on which key to use.
+        // The format of the hint must be agreed upon out of band by the
+        // signer and the verifiers, and so is not subject to this
+        // specification.
+        string hint = 1;
+}
+
+message X509Certificate {
+        // PEM encoded certificate(s), with indices 0 to n.
+        // For information on PEM encoding, see
+        // https://www.rfc-editor.org/rfc/rfc7468#section-2.
+        // The first certificate in the array
+        // must be the leaf certificate used for signing. Any intermediate
+        // certificates must be stored as offset 1 to n-1, and the root
+        // certificate at position n.
+        repeated string certificates = 1;
+}
+
+message Bundle {
+        // MUST be application/vnd.dev.sigstore.bundle.v0.1+json
+        // when encoded as JSON.
+        string media_type = 1;
+        optional VerificationData verification_data = 2;
+        oneof verification_material {
+                PublicKeyIdentifier public_key = 3;
+                X509Certificate x509_certificate = 4;
+        }
+        oneof attestation {
+                MessageSignature message_signature = 5;
+                io.intoto.Envelope attestation_dsse = 6;
```

Comment:
The DSSE envelope is generic over payload types, which is a little confusing.

What advice to we want to provide to clients? Do we want to be prescriptive? Maybe we should just say: "you should always check the payload type" or something in a comment

Reply:
I think we should be open to what data the dsse envelope, from this bundle's perspective I think it's enough to offer the capability to store a DSSE envelope. As you said, we should comment that the implementor would need to verify that the content type in the envelope is as expected.

Reply:
This is now renamed to be a generic dsse envelope. Can we resolve?